ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/EthereumWireProtocolConfiguration.java

```diff
@@ -0,0 +1,159 @@
+        DEFAULT_MAX_GET_RECEIPTS,
+        DEFAULT_MAX_GET_NODE_DATA);
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public int getMaxGetBlockHeaders() {
+    return maxGetBlockHeaders;
+  }
+
+  public int getMaxGetBlockBodies() {
+    return maxGetBlockBodies;
+  }
+
+  public int getMaxGetReceipts() {
+    return maxGetReceipts;
+  }
+
+  public int getMaxGetNodeData() {
+    return maxGetNodeData;
+  }
+
+  @Override
+  public boolean equals(final Object obj) {
+    if (!(obj instanceof EthereumWireProtocolConfiguration)) {
+      return false;
+    }
+    EthereumWireProtocolConfiguration other = ((EthereumWireProtocolConfiguration) obj);
+    return maxGetBlockHeaders == other.maxGetBlockHeaders
+        && maxGetBlockBodies == other.maxGetBlockBodies
+        && maxGetReceipts == other.maxGetReceipts
+        && maxGetNodeData == other.maxGetNodeData;
+  }
+
+  @Override
+  public int hashCode() {
+    return super.hashCode();
+  }
+
+  @Override
+  public String toString() {
+    return String.format(
+        "maxGetBlockHeaders=%s\tmaxGetBlockBodies=%s\tmaxGetReceipts=%s\tmaxGetReceipts=%s",
+        maxGetBlockHeaders, maxGetBlockBodies, maxGetReceipts, maxGetNodeData);
+  }
+
+  public static class Builder {
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xewp-max-get-headers"},
+        paramLabel = "<INTEGER>",
+        description =
+            "Maximum request limit for Ethereum Wire Protocol GET_BLOCK_HEADERS. (default: ${DEFAULT-VALUE})")
+    private PositiveNumber maxGetBlockHeaders =
+        PositiveNumber.fromInt(EthereumWireProtocolConfiguration.DEFAULT_MAX_GET_BLOCK_HEADERS);
+
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xewp-max-get-bodies"},
+        paramLabel = "<INTEGER>",
+        description =
+            "Maximum request limit for Ethereum Wire Protocol GET_BLOCK_BODIES. (default: ${DEFAULT-VALUE})")
+    private PositiveNumber maxGetBlockBodies =
+        PositiveNumber.fromInt(EthereumWireProtocolConfiguration.DEFAULT_MAX_GET_BLOCK_BODIES);
+
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xewp-max-get-receipts"},
+        paramLabel = "<INTEGER>",
+        description =
+            "Maximum request limit for Ethereum Wire Protocol GET_RECEIPTS. (default: ${DEFAULT-VALUE})")
+    private PositiveNumber maxGetReceipts =
+        PositiveNumber.fromInt(EthereumWireProtocolConfiguration.DEFAULT_MAX_GET_RECEIPTS);
+
+    @CommandLine.Option(
+        hidden = true,
+        names = {"--Xewp-max-get-node-data"},
+        paramLabel = "<INTEGER>",
+        description =
+            "Maximum request limit for Ethereum Wire Protocol GET_NODE_DATA. (default: ${DEFAULT-VALUE})")
+    private PositiveNumber maxGetNodeData =
+        PositiveNumber.fromInt(EthereumWireProtocolConfiguration.DEFAULT_MAX_GET_NODE_DATA);
+
+    public Builder maxGetBlockHeaders(final PositiveNumber maxGetBlockHeaders) {
```

Comment:
Could we change the method to just int, and then set the value to a PositiveNumber?  That reduces the friction for API consumers while preserving the error checking.